14.0 shopinvader WIP

sebastienbeau opened the pull request:
TODO
- [ ] merge this fix https://github.com/OCA/server-tools/pull/2002
- [x] merge this refactor https://github.com/shopinvader/odoo-shopinvader/pull/1170
- [ ] move this PR to shopinvader
- [ ] propose the sale-configurator project to oca

sebastienbeau committed:
[IMP] start adding configurator module for shopinavder

sebastienbeau committed:
[IMP] add categ of option

sebastienbeau committed:
[IMP] refactor code to avoid duplicated code

sebastienbeau committed:
[IMP] start adding services (extraction of existing custom)

sebastienbeau committed:
[REF] refactor code and start adding test

sebastienbeau committed:
[IMP] add area API

sebastienbeau committed:
[REF] refactor code for options

sebastienbeau committed:
[ADD] add module shopinvader_sale_configurator_variant

sebastienbeau committed:
[FIX] fix shopinvader_sale_configurator_variant getting name

sebastienbeau committed:
[REF] refactor code to get price info and all option info

sebastienbeau committed:
[FIX] fix shopinvader_sale_configurator_variants

sebastienbeau committed:
[FIX] fix perf issue

paradoxxxzero committed:
[ADD] sale_configurator_option_message, shopinvader_sale_configurator_option_message

paradoxxxzero commented on shopinvader_sale_configurator_option/models/product_product.py:
```diff
@@ -0,0 +1,36 @@
+    _inherit = "product.product"
+
+    shopinvader_price = fields.Serialized(
+        compute="_compute_shopinvader_price", string="Shopinvader Price"
```
This field label seems to conflict with the `price` field https://github.com/shopinvader/odoo-shopinvader/blob/14.0/shopinvader/models/shopinvader_variant.py#L60

(Getting warning: `Two fields (shopinvader_price, price) of [] have the same label: Shopinvader Price.`)

sebastienbeau committed:
[IMP] add test to ensure onchange playing to not do any mess

See PR https://github.com/akretion/sale-configurator/pull/40
sale_configurator_option was adding default option ids when calling product_option_id
this was doing a huge mess when the user log in (option was convert to normal line)
the fix is done their but by security we keep a test here

sebastienbeau committed:
[FIX] fix warnings

sebastienbeau commented:
@PierrickBrun j'ai quasi finir de corriger cette PR, (les tests passes en local mais on a un soucis avec les tests de sale_configurator_option quand shopinvader est installé)
Au passage je me suis rendu compte que j'ai perdu ce commit https://github.com/akretion/sale-configurator/commit/549b338c1b7680b1eeb449c9e356d7bfcedd2410 mais je sais pas pourquoi tu l'avais fait? (j'ai aucun test qui pète lié à ça)

sebastienbeau committed:
shopinvader_sale_configurator_variant: add test

sebastienbeau committed:
shopinvader_sale_configurator_variant: force price to 0 as onchange set it to normal price

sebastienbeau committed:
shopinvader_sale_configurator_option: fix price computation with pricelist without discount

sebastienbeau committed:
Merge pull request #29 from akretion/14.0-sale_configurator_option_message

[ADD] sale_configurator_option_message

sebastienbeau committed:
Fix build ci
